CodeEditModules/Modules/AppPreferences/src/Model/General/GeneralPreferences.swift

```diff
@@ -139,15 +146,24 @@ public extension AppPreferences {
         case hideAll
         case showAll
         case showOnly
+        case hideOnly
     }
 
-    /// The collection of displayed file extensions of ``FileExtensions/showOnly`` preference
-    struct ShowedFileExtensions: Codable, Hashable {
-        public var extensions: [String]
-        public static var `default` = ShowedFileExtensions(extensions: [
+    /// The collection of file extensions used by ``FileExtensions/showOnly`` or  ``FileExtensions/hideOnly`` preference
+    struct FileExtensionsCollection: Codable, Hashable {
+        public var extensions: [String] {
+            string
+                .components(separatedBy: ",")
+                .map({$0.trimmingCharacters(in: .whitespacesAndNewlines)})
+                .filter({!$0.isEmpty})
+        }
+
+        public var string: String
```

Comment:
I think the `string` should be computed (get, set) from the array instead of the other way around since the `string` is just a representation of the data.

Reply:
Yeah, i tried to use array as source for Binding, but it all ends up, to impossible redacting. Like commas and space symbols not deleting or adding.

Reply:
```swift
public var extensions: [String]

public var string: String {
  get {
    extensions.joined(separator: ", ")
  }
  set {
    extensions = string
                  .components(separatedBy: ",")
                  .map({$0.trimmingCharacters(in: .whitespacesAndNewlines)})
                  .filter({!$0.isEmpty})
  }
}

Reply:
Tried the same getter and setter, leading to that behaviour:

https://user-images.githubusercontent.com/36012972/167247556-88d8c31c-8d47-46e7-b4a3-82e350934912.mov

Reply:
Ah I see..

Reply:
An approach that works but in case of adding a comma an empty string gets added to the array. See the video below. Would that be a problem?

```swift
set {
    if newValue.count > string.count {
        extensions = newValue
            .components(separatedBy: ",")
            .map({$0.trimmingCharacters(in: .whitespacesAndNewlines)})
    } else {
        extensions = newValue
            .components(separatedBy: ",")
            .map({$0.trimmingCharacters(in: .whitespacesAndNewlines)})
            .filter({!$0.isEmpty})
    }
}
```

https://user-images.githubusercontent.com/9460130/167249726-e98c4949-a1ef-4466-b71e-81cb77bf3687.mov

Reply:
@MarcoCarnevali what's your take on this?

Reply:
I think that wouldn't be a problem, also I am considering this filter

```swift
filter({!$0.isEmpty || string.count < newValue.count })
```

Reply:
Yeah that's definitely a cleaner way 😉